src/plugins/usage_collection/server/collector/collector.ts

```diff
@@ -35,8 +35,8 @@ export interface SchemaField {
 }
 
 export type RecursiveMakeSchemaFrom<U> = U extends object
-  ? MakeSchemaFrom<U>
-  : { type: AllowedSchemaTypes };
+  ? { type?: 'object'; isArray?: boolean } & MakeSchemaFrom<U>
+  : { type: AllowedSchemaTypes; isArray?: boolean };
```

Comment:
Should we enforce `isArray` when it's an array? Maybe as part of the `MakeSchemaFrom` to avoid repetition?

```
export type RecursiveMakeSchemaFrom<U> = U extends object
  ? MakeSchemaFrom<U>
  : { type: AllowedSchemaTypes };

export type MakeSchemaFrom<Base> = {
  [Key in keyof Base]: Base[Key] extends Array<infer U>
    ? { isArray: true } & RecursiveMakeSchemaFrom<U>
    : RecursiveMakeSchemaFrom<Base[Key]>;
};
```

I'd say it would be even easier to type and spot at first glance if we define arrays as

```
export type MakeSchemaFrom<Base> = {
  [Key in keyof Base]: Base[Key] extends Array<infer U>
    ? { type: 'array', items: RecursiveMakeSchemaFrom<U> }
    : RecursiveMakeSchemaFrom<Base[Key]>;
};
```

Reply:
I kept it as optional until we figure out if we'll need it for the index mapping. ES does not care about arrays so this is just additional meta data about the reported data.

Reply:
I think we def need it for the indexer to know it (either for the extended or the aggregated views).